### PR TITLE
Decode percent encoded string parameters

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
## Checklist:

- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

**Comment by vadimeisenbergibm** [comment]
@shmuelk @ianpartridge please merge the PR

File: Sources/Kitura/RouterRequest.swift [hunk 1]
```diff
@@ -135,7 +135,19 @@ public class RouterRequest {
...
+            let valueReplacingPlus = value.replacingOccurrences(of: "+", with: " ")
+            if let decodedValue = valueReplacingPlus.removingPercentEncoding {
+                decodedParameters[parameter] = decodedValue
+            } else {
+                Log.warning("Unable to decode parameter \(parameter)")
+                decodedParameters[parameter] = valueReplacingPlus
+            }
+        }
+        return decodedParameters
+    }
```
Comment (shmuelk) [inline Sources/Kitura/RouterRequest.swift]: This code needs to be done differently. All of this processing can't happen every time one simply does request.queryParameters["plover"].

Also I think @na-gupta's changes are going to conflict here.....

File: Sources/Kitura/RouterRequest.swift [hunk 2]
```diff
@@ -155,6 +155,21 @@ public class RouterRequest {
         parsedURL = URLParser(url: serverRequest.url, isConnect: false)
         url = String(serverRequest.urlString)
         headers = Headers(headers: serverRequest.headers)
+        queryParameters = RouterRequest.decode(queryParameters: parsedURL.queryParameters)
```
Comment (vadimeisenbergibm) [inline Sources/Kitura/RouterRequest.swift]: @shmuelk is this implementation OK ? made `queryParameters` let variable, which is set in `init()`

File: Sources/Kitura/RouterRequest.swift [hunk 3]
```diff
@@ -135,7 +135,19 @@ public class RouterRequest {
     public internal(set) var parameters: [String:String] = [:]
 
     /// List of query parameters.
-    public var queryParameters: [String:String] { return parsedURL.queryParameters }
+    public lazy var queryParameters: [String:String] = { [unowned self] in
```
Comment (naithar) [inline Sources/Kitura/RouterRequest.swift]: You are welcome.

**Comment by vadimeisenbergibm** [comment]
@shmuelk please merge the PR.

**Pull request merged**